### ISSUE TITLE
avatar names on stage

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/limelight#204]
it should be possible for avatar names to appear under the avatar image on the stage; players should be able to choose whether to show or hide this name on the stage.

**Comment by JoeIceCream** [comment]
@rohrlach Hey Paul, I've had a go at helping out Kit's issue, but we're running into the same problem that I had before where we can't get the data to reach the method under "name (data) ->" in the avatar.coffee file. Would there be any chance you could take a look and see why the data.avatar_name isn't receiving the data from the avatar_channel.rb file? Thanks

**Comment by rohrlach** [comment]
It's because on line 72 of avatars.coffee you are calling the name function like this:

```when 'name' then name data.avatar_name```

IE: passing the name directly to the function.

But the function ```name``` on line 44 is expecting an object ```data``` on which to find an ```avatar_name``` attribute:

```
console.log("running name, data.avatar_name = " + data.avatar_name)
```

A couple of breakpoints in your browser's debugger would tell you this - Please use the tools at your disposal.

**Comment by blankgaming** [comment]
![btn](https://user-images.githubusercontent.com/22442427/30142778-b25e69a0-93d6-11e7-8e40-e355e607cdfe.PNG)
![btnisnull](https://user-images.githubusercontent.com/22442427/30142779-b28955fc-93d6-11e7-93ef-d24a5df82ee3.PNG)

@rohrlach Hi Paul, I had a go again this morning. I found that the btn class for the avatar name button is appearing null after getting the query for it as you can see on the screenshots. Even I logged it and check what is happening still not seeing why is null, please have a look as I may have tunnel visioned

**Comment by rohrlach** [comment]
@blankgaming it's ```null``` because your selector query string doesn't match anything.  Look at the log output of ```data``` - You'll notice that it doesn't have an ```avatar_id``` attribute and so ```#{data.avatar_id}``` (in your selector query) is null.

**Comment by blankgaming** [comment]
![btnisnull](https://user-images.githubusercontent.com/22442427/30143611-e253b3dc-93da-11e7-8db1-f48141cdfd52.PNG)

**Comment by rohrlach** [comment]
So you're good now?

**Comment by blankgaming** [comment]
Yes for now, Im left with to toggle to whether to display or hide the avatar name(s).

**Comment by rohrlach** [comment]
Good stuff 👍

**Comment by blankgaming** [comment]
![capture](https://user-images.githubusercontent.com/22442427/30256885-1856af50-9702-11e7-905f-a75a96ce1084.PNG)


Hi Paul @rohrlach ,
I continued working on issue 204 after Joe left for his class, I was able to display the avatar name in the middle since it was intended. However each time I move the avatar image to another location in the canvas the name disappear until I click the show/hide button again then it will show. Please have a look if this approach is ideal.

**Comment by rohrlach** [comment]
@blankgaming that doesn't sound particularly ideal Look at how the avatar movement is implemented.  You'll notice that in order to accomplish it, the entire stage needs to be redrawn.

This means that the state of the stage needs to be persisted between redraws (Which it is)

**Comment by blankgaming** [comment]
![show](https://user-images.githubusercontent.com/22442427/30361935-ab0774b8-98ad-11e7-980c-b5114829bb07.PNG)

![hide](https://user-images.githubusercontent.com/22442427/30361940-b1abff14-98ad-11e7-855b-732598679d06.PNG)

Hi Paul @rohrlach , The avatar name was able to be shown or hidden when clicking the "show/hide button" however it's only working when you place the avatar again on the canvas. I've tried to add a name function to pass the value of the button to the cable.js, but for some reasons, it's not sending it through the name function created. It's only possible through the place function. I've attached the images and uploaded the code, please have a look. Thanks

**Comment by rohrlach** [comment]
@blankgaming I fail to understand the point of ```cable.js``` all you're meant to be doing is adding a label to the existing avatar behaviour.

To your question - It's probably because all your ```name``` action doesn't send anything, nor does the wrapping function receive anything:
```
name: () ->
      @perform 'name'
```
The function doesn't accept any parameters and when the websocket action is fired, no additional information is send other than the type of action to be performed meaning that ```data``` is nil here:
```
def name(data)
    unless current_user.nil? || @avatar_allocation[data['avatar_id']] != nil
      AvatarChannel.broadcast_to @stage, {  
         action: 'name', 
         avatar_name: data['avatar_name']
       }
  end
end
```

**Comment by helenvarley** [comment]
this is working, but the font is way too small. also on one avatar, the name stays close to the avatar despite the size, but on my (distorted/squished) helen avatar, the name is a long way away from the actual image.

**Comment by charlie0605** [comment]
is this issue same as #305 ?

**Comment by thongteav** [comment]
I believe so, and I've fixed the distortion bug. So the name should be right below the image now.

**Comment by rohrlach** [comment]
@helenvarley can you confirm?

**Comment by helenvarley** [comment]
confirmed. however the font of the name is very small - on my screen it's almost too small to read. could it be slightly bigger?

**Comment by charlie0605** [comment]
@rohrlach this issue is already been fixed in #305 (PR #342), can this issue be close or is there anymore needs to be done here?
